api/v1/serializers/students-serializer.js

```diff
@@ -0,0 +1,270 @@
+};
+
+const serializeGrades = (rawGrades, osuID) => {
+  const serializerArgs = getSerializerArgs(osuID, 'GradesResult', 'grades', false);
+
+  _.forEach(rawGrades, (rawGrade) => {
+    rawGrade.creditHours = parseFloat(rawGrade.creditHours);
+  });
+
+  return getJSONAPISerializer(serializerArgs, rawGrades);
+};
+
+const serializeClassSchedule = (rawClassSchedule, osuID) => {
+  const serializerArgs = getSerializerArgs(osuID, 'ClassScheduleResult', 'class-schedule', false);
+  const rawDataByTermAndCRN = {};
+
+  _.forEach(rawClassSchedule, (rawRow) => {
+    const id = `${osuID}-${rawRow.term}-${rawRow.courseReferenceNumber}`;
+
+    rawDataByTermAndCRN[id] = rawDataByTermAndCRN[id] || { faculty: [], meetingTimes: [] };
+
+    const faculty = {
+      osuID: rawRow.facultyOSUID,
+      name: rawRow.facultyName,
+      email: rawRow.facultyEmail,
+      primary: rawRow.facultyPrimary === 'Y',
+    };
+
+    const meetingTime = {
+      beginDate: rawRow.beginDate,
+      beginTime: fourDigitToTime(rawRow.beginTime),
+      endDate: rawRow.endDate,
+      endTime: fourDigitToTime(rawRow.endTime),
+      room: rawRow.room,
+      building: rawRow.building,
+      buildingDescription: rawRow.buildingDescription,
+      campus: rawRow.campus,
+      hoursPerWeek: parseFloat(rawRow.hoursPerWeek),
+      creditHourSession: parseFloat(rawRow.creditHourSession),
+      scheduleType: rawRow.meetingScheduleType,
+      scheduleDescription: rawRow.meetingScheduleDescription,
+      weeklySchedule: _.without([
+        rawRow.monday ? 'M' : null,
+        rawRow.tuesday ? 'T' : null,
+        rawRow.wednesday ? 'W' : null,
+        rawRow.thursday ? 'Th' : null,
+        rawRow.friday ? 'F' : null,
+        rawRow.saturday ? 'Sa' : null,
+        rawRow.sunday ? 'Su' : null,
+      ], null),
+    };
+
+    rawDataByTermAndCRN[id] = {
+      identifierField: id,
+      academicYear: rawRow.academicYear,
+      academicYearDescription: rawRow.academicYearDescription,
+      courseReferenceNumber: rawRow.courseReferenceNumber,
+      courseSubject: rawRow.courseSubject,
+      courseSubjectDescription: rawRow.courseSubjectDescription,
+      courseNumber: rawRow.courseNumber,
+      courseTitle: rawRow.courseTitleLong || rawRow.courseTitleShort,
+      sectionNumber: rawRow.sectionNumber,
+      term: rawRow.term,
+      termDescription: rawRow.termDescription,
+      scheduleDescription: rawRow.scheduleDescription,
+      scheduleType: rawRow.scheduleType,
+      creditHours: parseFloat(rawRow.creditHours),
+      registrationStatus: rawRow.registrationStatus,
+      gradingMode: rawRow.gradingMode,
+      continuingEducation: rawRow.continuingEducation === 'Y',
+      faculty: uniqConcat(rawDataByTermAndCRN[id].faculty, faculty),
+      meetingTimes: uniqConcat(rawDataByTermAndCRN[id].meetingTimes, meetingTime),
+    };
+  });
+
+  const newClassSchedule = [];
+  _.forEach(rawDataByTermAndCRN, (rawData) => {
+    newClassSchedule.push(rawData);
+  });
```

Comment:
Could this be replaced with something like this?
```js
const newClassSchedule = _.clone(rawDataByTermAndCRN);
```
I'm not completely sure on this, but this seems like it could work based on the documentation for `_.clone()` https://lodash.com/docs/4.17.11#clone

Reply:
From our discussion, `newClassSchedule` is an array that is taking each value in the `rawDataByTermAndCRN` object, so clone would not work here. `_.forEach` seems to be the best solution so I don't think any changes are needed